src/dialogs/more-info/ha-more-info-tab-history.ts

```diff
@@ -85,7 +85,18 @@ export class MoreInfoTabHistoryDialog extends LitElement {
       clearInterval(this._historyRefreshInterval);
     }
 
-    if (changedProps.has("entityId")) {
+    if (!this.hass) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant;
+    const newStateObj = this.hass.states[this.entityId];
+    const oldStateObj = oldHass?.states[this.entityId];
+
+    if (
+      changedProps.has("entityId") ||
+      (oldHass && newStateObj !== oldStateObj)
+    ) {
```

Comment:
We don't want to fetch all history again on every state change, what is wrong with what we have now? We fetch history every minute, I think is the same as before?

Reply:
We could add updating our logbook. And we could limit updating to when the state changed so if the state is changed and a minute is passed update.

Reply:
Yea... I think maybe I just need to add log book in there. But I do feel like if I change the state in the more info and look over it should update

Reply:
If an entity changes a lot we can't keep updating the history